plugins/gcp/resoto_plugin_gcp/resources/compute.py

```diff
@@ -2713,10 +2719,43 @@ class GcpInstance(GcpResource, BaseInstance):
     instance_start_restricted: Optional[bool] = field(default=None)
     instance_status_message: Optional[str] = field(default=None)
     instance_tags: Optional[GcpTags] = field(default=None)
+    instance_type: Optional[str] = None
+    instance_cores: float = 0.0
+    instance_memory: float = 0.0
+    _machine_type_link: Optional[str] = field(default=None, alias="_machine_type_link")
+    _machine_type: Optional[BaseInstanceType] = field(default=None, alias="_machine_type")
```

Comment:
why do we need this - can we remove it?

Reply:
It does not make sense to check if `self._machine_type` is an instance of `BaseInstanceType` when this property is not there/not this.
See [here](https://github.com/someengineering/resoto/blob/5987d3678789d4a6c48be60ec5345baac873dbd3/plugins/gcp/resoto_plugin_gcp/resources/compute.py#L2728-L2733): This is the part we agreed on keeping because it defines the `BaseInstance`